src/core/abi/errors.ts

```diff
@@ -0,0 +1,20 @@
+export class ValidationError extends Error {
```

Comment:
I'm assuming these error classes are to match the rust behaviour?
There certainly a difference in how the errors are unwrapped, so might be worth a quick chat.